### DISCUSSION TITLE
Is it possible to add custom api endpoints ?

### DISCUSSION THREAD

**Discussion by nazarDzhuruk** [koltyakov/sp-rest-proxy#165] [Q&A]
Is it possible to add custom api endpoints ? For example: I want to create GET endpoint '/test'  which will trigger any GET existing endpoint of sp-rest-proxy and then send response data via existing POST method ?

**Comment by koltyakov** [comment]
Hi @nazarDzhuruk,

Not getting your idea. If you have a custom web service hosted in On-premise SharePoint you can reach it via sp-rest-proxy if the relative pass is satisfied, even static resources or legacy web services should work this way.

But what a sort of transformations from GET to POST? Why? How will it work in a runtime?

sp-rest-proxy is a tool for development time. Instead of deploying an app to SharePoint a developer can consume the API from within a local workbench. And the app code should be identical if it was running in SharePoint page context, otherwise it won't work when deployed.
